File: shared/src/commonMain/kotlin/com/cachekid/companion/shared/MissionModels.kt
package com.cachekid.companion.shared

import kotlinx.serialization.Serializable

@Serializable
data class MissionTarget(
    val latitude: Double,
    val longitude: Double,
)

@Serializable
data class MissionWaypoint(
    val latitude: Double,
    val longitude: Double,
)

@Serializable
data class ActiveMission(
    val missionId: String,
    val cacheCode: String,
    val sourceTitle: String,
    val childTitle: String,
    val summary: String,
    val target: MissionTarget,
    val routeOrigin: MissionTarget? = null,
    val waypoints: List<MissionWaypoint> = emptyList(),
)

@Serializable
data class LocationSnapshot(
    val latitude: Double,
    val longitude: Double,
)

@Serializable
data class MissionPackage(
    val mission: ActiveMission,
    val routePoints: List<MissionWaypoint>,
    val createdAt: Long,
    val version: String = "1.0",
)

File: ios-host/CacheKidHostApp/MissionModels.swift
import Foundation

/// Represents a geocache target position
struct MissionTarget: Codable, Equatable {
    let latitude: Double
    let longitude: Double
}

/// A waypoint along the route
struct MissionWaypoint: Codable, Equatable {
    let latitude: Double
    let longitude: Double
}

/// Full mission definition shared between host and kid
struct ActiveMission: Codable, Identifiable {
    var id: String { missionId }
    let missionId: String
    let cacheCode: String
    let sourceTitle: String
    let childTitle: String
    let summary: String
    let target: MissionTarget
    let routeOrigin: MissionTarget?
    let waypoints: [MissionWaypoint]
}

/// Snapshot of player location
struct LocationSnapshot: Codable {
    let latitude: Double
    let longitude: Double
}

/// Package sent from host to kid device
struct MissionPackage: Codable {
    let mission: ActiveMission
    let routePoints: [MissionWaypoint]
    let createdAt: Date
    let version: String
}
